libvmi/driver/xen/xen_events.c

```diff
@@ -862,7 +862,10 @@ status_t xen_set_mem_access(vmi_instance_t vmi, mem_access_event_t *event,
     dbprint(VMI_DEBUG_XEN, "--Setting memaccess for domain %"PRIu64" on physical address: %"PRIu64" npages: %"PRIu64"\n",
         dom, event->physical_address, npages);
 
-    rc = xc_set_mem_access(xch, dom, access, page_key, npages);
+    if ( !altp2m_idx )
```

Comment:
Also see http://xenbits.xen.org/gitweb/?p=xen.git;a=blob;f=xen/arch/x86/mm/p2m-ept.c#l854
When you change permissions on the hostp2m (0) it gets automatically propagated to all altp2m views as well.